### PR TITLE
Build pylance on Mac and Update python build instruction.

### LINKED ISSUE #14: Add Readme and setup.py for Python package
The readme should include:

* How to prepare `pyarrow` 
* How to build `pylance`.
* How to test
* Code style and etc.

### PR DESCRIPTION
**Pull request by eddyxu**
Closes #14

### CODE REVIEW

File: python/thirdparty/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Build apache arrow
+function build_arrow {
+    git clone git@github.com:apache/arrow
+    pushd arrow
+    git checkout ${APACHE_ARROW_VERSION}
+    git submodule update --init
+    pushd python
+    pip install -r requirements-build.txt
+
+    export OPENSSL_ROOT_DIR=/opt/homebrew/opt/openssl@1.1
```
Comment (darcy-shen) [inline python/thirdparty/build.sh:16]: what's the diff between `openssl@1.1` and `openssl@3`

Comment (eddyxu) [reply python/thirdparty/build.sh:16]: Not sure. This is the one installed on my system. Would be great to make CMake does not require these exports on Mac.

File: python/setup.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    name="pylance",
+    version="0.0.1",
+    author="Lance Developers",
+    author_email="contact@eto.ai",
+    description="Python extension for lance",
+    long_description="",
+    ext_modules=cythonize(extensions),
+    zip_safe=False,
+    extras_require={"test": ["pytest>=6.0"]},
+    python_requires=">=3.9",
```
[the remark this answers is not in the document]
Comment (changhiskhan) [reply python/setup.py:41]: python 3.7 security support is ending in 11 months, I don't think it's worth it for a new project to taken on the extra baggage

[the remark this answers is not in the document]
Comment (changhiskhan) [reply python/setup.py:41]: but reasonable to support 3.8+

File: python/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+pushd python
+source ${HOME}/.venv/nft/bin/activate
+python setup.py develop
+```
+
+Test the installation in python:
+
+```python
+import duckdb
+import pylance
```
Comment (changhiskhan) [inline python/README.md:60]: i wonder if ppl would get confused with the MSVC thing?

**Review by changhiskhan — APPROVED** [review]

File: python/setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (eddyxu) [reply python/setup.py]: fixed

**Pull request merged**